akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -124,4 +146,30 @@ object GraphStages {
       (logic, cancellable)
     }
   }
+
+  /**
+   * INTERNAL API.
+   *
+   * This source is not reusable, it is only created internally.
+   */
+  private[stream] class MaterializedValueSource[T](val computation: MaterializedValueNode, val out: Outlet[T]) extends GraphStage[SourceShape[T]] {
```

Comment:
Looks ok. Maybe there are cheaper ways to do this, but since these sources are rare, and the cost is only paid when it is used, I think this is fine.